billy-portugal/src/main/java/com/premiumminds/billy/portugal/services/documents/PTGenericInvoiceIssuingHandler.java

```diff
@@ -155,16 +158,14 @@ private void validateSeriesHasNoWhiteSpaces(final String series) throws Document
 		}
 	}
 
-	private InvoiceSeriesEntity getInvoiceSeries(final T document, String series, LockModeType lockMode) {
+	private InvoiceSeriesEntity getInvoiceSeries(final T document, String series, LockModeType lockMode)
+		throws DocumentSeriesDoesNotExistException
+	{
         InvoiceSeriesEntity invoiceSeriesEntity =
                 this.daoInvoiceSeries.getSeries(series, document.getBusiness().getUID().toString(), lockMode);
 
         if (null == invoiceSeriesEntity) {
-            InvoiceSeriesEntity entity = new JPAInvoiceSeriesEntity();
-            entity.setBusiness(document.getBusiness());
-            entity.setSeries(series);
-
-            invoiceSeriesEntity = this.daoInvoiceSeries.create(entity);
+            throw new DocumentSeriesDoesNotExistException(series);
```

Comment:
At the moment we are currently using the series as the only content of the message on `DocumentSeriesDoesNotExistException` instances, can we provide a more complete contextual message?